### PR TITLE
Fix remove case from group if case has been already removed in another tab

### LINKED ISSUE #4205: Error in remove_case_group
```
2023-11-13 09:44:43,379 - [scout.server.app](http://scout.server.app/) - ERROR: Exception on /cust087/F0044881-2/6551e0ef1a1a70ed7468a5b6/remove_case_group [GET] [in /venv/lib/python3.8/site-packages/flask/app.py:1744]
Traceback (most recent call last):
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python3.8/site-packages/flask_cors/extension.py", line 176, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/worker/app/scout/server/blueprints/cases/views.py", line 910, in remove_case_group
    controllers.remove_case_group(store, current_user, institute_id, case_name, case_group)
  File "/home/worker/app/scout/server/blueprints/cases/controllers.py", line 1075, in remove_case_group
    current_group_ids.remove(ObjectId(case_group))
ValueError: list.remove(x): x not in list
```

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #4205 
Specifically:
-  "x" button to remove a case from a group sometimes is not visible, based on the clicked status:

image

- Fix remove the case from a group whenever it has been already removed in another browser tab (fix #4205 )

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Locally, from demo case 643594, create a group and add cancer_case to this group. Go o another browser tab and open case 643594. Remove cancer_case from the group and go to the other tab. Try to remove cancer_case from the group also in the pther tab

**Expected outcome**:
- Main branch should crash
- This branch should just reload the case page

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/cases/controllers.py [hunk 1]
```diff
@@ -1071,8 +1071,10 @@ def remove_case_group(store, current_user, institute_id, case_name, case_group):
 
     current_group_ids = case_obj.get("group", [])
 
-    # STR OBJID mismatch?
-    current_group_ids.remove(ObjectId(case_group))
+    obj_id_group = ObjectId(case_group)
+    if obj_id_group not in current_group_ids:
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py:1075]: 💯

File: scout/server/blueprints/cases/templates/cases/case.html [hunk 2]
```diff
@@ -356,7 +356,7 @@ <h6 class="mt-3 ms-3">
                   <a class="text-dark" href="{{ url_for('cases.case', institute_id=grouped_case.owner, case_name=grouped_case.display_name) }}">{{ grouped_case.display_name }}</a>
 
                   <a href="{{ url_for('cases.remove_case_group', institute_id=institute._id, case_name=grouped_case.display_name, case_group=group_id) }}" class="btn btn-link btn-sm">
-                      <span class="fa fa-remove"></span></a>
+                      <span class="fa fa-remove text-black"></span></a>
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/case.html:359]: Well spotted! They are visible on dark mode in both cases, but will still be now, so much better. Dark or black are both fine, just style.

Comment (northwestwitch) [inline scout/server/blueprints/cases/templates/cases/case.html:359]: i'm not really satisfied by this. Perhaps just remove the btn-link from the line above?

**Review by dnil — APPROVED** [review]

File: scout/server/blueprints/cases/templates/cases/case.html (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/case.html]: Also sounds like a plan!

**Comment by northwestwitch** [comment]
I think we should fix also cohort tags style to match the dark text here, because otherwise they don't look so nice, being so close and all:

image

**Comment by dnil** [comment]
> I think we should fix also cohort tags style to match the dark text here, because otherwise they don't look so nice, being so close and all:
> 
> image

Agreed!

**Comment by northwestwitch** [comment]
They look the same now:

image

**Pull request merged**